array-manipulation-3/src/zip.js

```diff
@@ -21,20 +20,16 @@ function zip(first, second) {
   var result = [];
   var container = [];
 
-  if (first.length < second.length) {
-    for (var i = 0; i < first.length; i++) {
-      container.push(first[i]);
-      container.push(second[i]);
-      checkContainer();
-    }
-  } else {
-    for (var z = 0; z < second.length; z++) {
-      container.push(first[z]);
-      container.push(second[z]);
-      checkContainer();
-    }
-  }
+  var smallestLength = Math.min(first.length, second.length);
+  if (first.length === smallestLength) {
+    var smallestParam = first;
+  } else smallestParam = second;
 
+  for (var i = 0; i < smallestParam.length; i++) {
```

Comment:
Use the `smallestLength` in your `for` loop here. Then you don't need `smallestParam`